plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -246,12 +245,31 @@ func compareTopics(expected []string, incoming []string) bool {
 }
 
 func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Message) error {
-	byteCount := len(string(msg.Payload())) +
-		len(string(rune(msg.MessageID()))) +
-		len(msg.Topic()) +
-		int(unsafe.Sizeof(msg.Duplicate())) +
-		len(string(msg.Qos())) +
-		int(unsafe.Sizeof(msg.Retained()))
+	var remainingLength int
+	var qosFlagsSize int
+	topicSize := len(msg.Topic()) + 2
+	qOsSize := 0
+	if msg.Qos() == 1 || msg.Qos() == 2 {
```

Comment:
This line of code where I try to access msg.QoS(), as well as the one below on line 267, is what's causing tests to fail, not sure why though. Tried converting to int or string as well, didn't work

Reply:
Test needs to be updated to implement the mock results for these methods as well: https://github.com/influxdata/telegraf/blob/0fced8872a1b321d21d0ed2608dd8349d2731baf/plugins/inputs/mqtt_consumer/mqtt_consumer_test.go#L165-L167